addons/auth_totp/models/res_users.py

```diff
@@ -51,8 +51,10 @@ def _mfa_url(self):
 
     @api.depends('totp_secret')
     def _compute_totp_enabled(self):
-        for r, v in zip(self, self.sudo()):
-            r.totp_enabled = bool(v.totp_secret)
+        self.env.cr.execute('SELECT id, totp_secret FROM res_users WHERE id = ANY(%s)', (list(self.ids),))
+        result = {_id: secret for _id, secret in self.env.cr.fetchall()}
+        for rec in self:
+            rec.totp_enabled = bool(result[rec.id])
```

Comment:
Even if the code works, here's another way of doing it:

pseudo code:
```
self.totp_secret = False
execute('SELECT id FROM res_users WHERE id in <ids> and totp_secret', self.ids)
ids = fetch
self.filtered(ids).totp_secret = True
```
It's your choice